### DISCUSSION TITLE
Running Javascipt queries from Deno takes over one  minute to return very basic query

### DISCUSSION THREAD

**Discussion by keef-ui** [sqlitecloud/discussions#45] [SDK]
`I` am running deno version 1.46.2. I run a basic script to query : 

```
import { Database } from 'npm:@sqlitecloud/drivers'

let database = new Database("sqlitecloud://user:password@xxx.sqlite.cloud:8860/chinook.db")

let name = 'Breaking The Rules'

let results = await database.sql`SELECT * FROM tracks WHERE name = ${name}`
console.log(results)

```  

 the query returns about one minute later with the result. But if I run the similar basic script in Node.js, the result comes back  within seconds.  My question is , has anyone tested the database module with Deno, if so what is the expected outcome?

**Comment by Gioee** [comment]
Hi @keef-ui,
Is this the project you're having issues with https://github.com/keef-ui/sqlite-deno ?
I've tried to run your project and as you can see it runs fine.
You can even try the _with-plain-javascript_ and _with-javascript-express_ examples inside the [example folder in the sqlitecloud-js repository](https://github.com/sqlitecloud/sqlitecloud-js/tree/main/examples) and they also run fine to me with Deno.
The Deno version I'm using is

> deno 1.46.3 (stable, release, aarch64-apple-darwin)
v8 12.9.202.5-rusty
typescript 5.5.2

https://github.com/user-attachments/assets/e47af6df-4ab3-46fd-9f89-2902f53e1646

Do you have more details about your issue?
Otherwise let me know if you need help with anything else :)

**Comment by keef-ui** [comment]
I am sorry, I think the problem is not with the queries, What i think is happening is that that the initial connection/authentication to DB is taking about 1 minute. But once its connected the , the queries will be  executed instantly and fast. I tested the following code with 2 queries, once CloudDb was authenticated, both the queries came back instantly.

```
import { Database } from 'npm:@sqlitecloud/drivers'
import "jsr:@std/dotenv/load";

const connectionString = Deno.env.get("SQLITECLOUD_URL")!;

let database = new Database(connectionString);

let name = 'Breaking The Rules'

let results = await database.sql`SELECT * FROM tracks WHERE name = ${name}`
// => returns [{ AlbumId: 1, Name: 'Breaking The Rules', Composer: 'Angus Young... }]
console.log(results)

let another_results = await database.sql`SELECT * FROM tracks WHERE name = ${name}`
console.log(another_results)

```

If you can let me know if there is anything else I can check in my end, please let me know.
I will update Deno to latest version and see if there is any difference also.

Thanks for looking into the problem!

**Comment by Gioee** [comment]
Hi @keef-ui,
This example also runs instantly and fine on my end. I can't notice anything wrong with it, the authentication happens in milliseconds.
Are you sure your internet connection is working correctly?
I've seen you were using an old version of our driver and I've tried to update it to the latest version which also runs fine.
Would you like to share with me your project id or your connection string **without** the apikey? So we can check if there's anything wrong with your cluster! Thanks.

**Reply by keef-ui** [reply]
I can convert the same code to Run on Node.js and it works fine: ie Node.js code I tested was : 
```
import { Database } from '@sqlitecloud/drivers';



let database = new Database(MY_CONNECTION_STRING_GOES_HERE);
let name = 'Breaking The Rules';


(async function main () {
    let results = await database.sql`SELECT * FROM tracks WHERE name = ${name}`
    console.log(results)
})();

// => returns [{ AlbumId: 1, Name: 'Breaking The Rules', Composer: 'Angus Young... }]
```
So network is working fine.

My connection string is sqlitecloud://cmfjiktliz.sqlite.cloud:8860/chinook.sqlite

**Reply by Gioee** [reply]
@keef-ui what's the operating system you're using to run Deno?

**Reply by keef-ui** [reply]
I have run Deno on Windows 11 pro, I also ran it on my laptop (Windows 10 Home version), both have same outcome

**Reply by Gioee** [reply]
Thanks, I'll try with Windows as soon as I can. Meanwhile would you like to try this code with Deno?
```
import { Database } from 'npm:@sqlitecloud/drivers'
import "jsr:@std/dotenv/load";

const connectionString = Deno.env.get("SQLITECLOUD_URL")!;

let database = new Database({
    connectionstring: connectionString,
    usewebsocket: true,
    verbose: true
});

let name = 'Breaking The Rules'

let results = await database.sql`SELECT * FROM tracks WHERE name = ${name}`
// => returns [{ AlbumId: 1, Name: 'Breaking The Rules', Composer: 'Angus Young... }]
console.log(results)

let another_results = await database.sql`SELECT * FROM tracks WHERE name = ${name}`
console.log(another_results)

```

**Reply by keef-ui** [reply]
Great! its returning results instantly, but in the console.log message it prints out a warning before returning the result object :
**"Warning: Not implemented: ClientRequest.options.createConnection"** :

```
Warning: Not implemented: ClientRequest.options.createConnection
Warning: Not implemented: ClientRequest.options.createConnection
SQLiteCloudRowset(1) [
  SQLiteCloudRow {
    TrackId: 12,
    Name: "Breaking The Rules",
    AlbumId: 1,
    MediaTypeId: 1,
    GenreId: 1,
    Composer: "Angus Young, Malcolm Young, Brian Johnson",
    Milliseconds: 263288,
    Bytes: 8596840,
    UnitPrice: 0.99
  }
]
SQLiteCloudRowset(1) [
  SQLiteCloudRow {
    TrackId: 12,
    Name: "Breaking The Rules",
    AlbumId: 1,
    MediaTypeId: 1,
    GenreId: 1,
    Composer: "Angus Young, Malcolm Young, Brian Johnson",
    Milliseconds: 263288,
    Bytes: 8596840,
    UnitPrice: 0.99
  }
]

```

**Comment by Gioee** [comment]
Hi @keef-ui, after further investigation Deno 2 seems to be working with @sqlitecloud/drivers in Windows without using websockets. Can you confirm that it also works on your machine? So I can close this issue.

Here you can see that it works in our CI by running the examples we have in the examples folder.

https://github.com/sqlitecloud/sqlitecloud-js/actions/runs/11963531862/job/33354141189

**Comment by keef-ui** [comment]
I still have to use websockets, it still a problem without the WebSocket settings. I have setup a temporary api key so you can test with my sqlite cloud account, so that we can eliminate that. I think there was 2 emails associated with my account one for gmail.com and one for googlemail.com.  I will send the api key for you to test through the sqlitecloud dashboard contact form.

**Comment by Gioee** [comment]
@keef-ui thanks I've received the apikey. It works with it on MacOS. On Monday I'll try also on Windows.

**Comment by Gioee** [comment]
@keef-ui I've found a way to reproduce it. I'll let you know as soon as I fix it.

**Reply by keef-ui** [reply]
Thanks for your help
